src/animals/Lion.ts

```diff
@@ -14,31 +16,18 @@ export default class Lion extends Mammal implements ICanRun, ICanTalk {
   talk(): void {
     console.log(`${this.name} is talking`);
   }
-  giveBirth(): void {
-    if (this.isMale) {
-      console.log(
-        `Only Females can give birth and ${this.name} is a proud male ${this.constructor.name}`
-      );
-    } else if (this.home === undefined) {
-      console.log(
-        `${this.name} can't give birth because she doesn't have a home to meet a male`
-      );
-    } else {
-      this.timesGivenBirth += 1;
-      for (let i = 1; i <= 5; i++) {
-        const num = Math.floor(Math.random() * (2 - 1 + 1)) + 1;
-        const name: string =
-          this.name + "-Baby-" + this.timesGivenBirth + "-" + i;
-        const gender: boolean = num === 1;
-        const age: number = 0;
-        const babyLion = new Lion(name, age, gender);
+
+  public giveBirth(): void {
+    const candidateLions: AnimalCandidate[] = super.giveBirth() as AnimalCandidate[];
+    if (candidateLions) {
+      candidateLions.forEach(lionObjectInfo => {
+        const babyLion = new Lion(lionObjectInfo.name, 0, lionObjectInfo.isMale);
         families.forEach((family) => {
```

Comment:
Here i believe we can avoid a nested for loop if for example before we go through all the canditateLions we find the family and add each baby lion there

Reply:
fixed it